pyuvdata/uvdata/mir_parser.py

```diff
@@ -923,36 +1106,640 @@ def _apply_tsys(self, jypk=130.0):
+                try:
+                    self._downselect_data(
+                        select_vis=False, select_raw=False, select_auto=True
+                    )
+                    load_auto = False
+                except MirMetaError:
+                    if allow_downselect:
+                        warnings.warn("Cannot downselect auto-correlation data.")
+
+        # Finally, if we didn't downselect or convert, load the data from disk now.
+        if load_vis or load_raw:
+            data_dict = self.read_data(
+                "cross",
+                return_vis=load_vis,
+                use_mmap=use_mmap,
+                read_only=read_only,
+            )
+
+            setattr(self, "vis_data" if load_vis else "raw_data", data_dict)
+
+            if load_vis:
+                # Since we've loaded in "fresh" data, we mark that tsys has
+                # not yet been applied (otherwise apply_tsys can thrown an error).
+                self._tsys_applied = False
+
+                # Apply tsys if needed.
+                if apply_tsys and load_vis:
+                    self.apply_tsys()
+
+        # We wrap the auto data here in a somewhat special way because of some issues
+        # with the existing online code and how it writes out data. At some point
+        # we will fix this, but for now, we triage the autos here. Note that if we
+        # already have the auto_data loaded, we can bypass this step.
+        if load_auto:
+            self.auto_data = self.read_data(
+                "auto", return_vis=False, use_mmap=use_mmap, read_only=read_only
+            )
+
+    def unload_data(self, unload_vis=True, unload_raw=True, unload_auto=True):
+        """
+        Unload data from the MirParser object.
+
+        Unloads the data-related attributes from memory, if they are loaded. Because
+        these attributes can be formidible in size, this operation will substantially
+        reduce the memory footprint of the MirParser object.
+
+        Parameters
+        ----------
+        unload_vis : bool
+            Unload the visibilities stored in the `vis_data` attribute, if loaded.
+            Default is True.
+        unload_raw : bool
+            Unload the raw visibilities stored in the `raw_data` attribute, if loaded.
+            Default is True.
+        unload_auto : bool
+            Unload the auto-correlations stored in the `auto_data` attribute, if loaded.
+            Default is True.
+        """
+        if unload_vis:
+            self.vis_data = None
+            self._tsys_applied = False
+        if unload_raw:
+            self.raw_data = None
+        if unload_auto:
+            self.auto_data = None
+
+    def _update_filter(self, update_data=None):
+        """
+        Update MirClass internal filters for the data.
+
+        Note that this is an internal helper function which is not for general user use,
+        but instead is part of the low-level API for the MirParser object. Updates
```

Comment:
"API" comment for internal methods again.

Reply:
Changed in a2f46b4b.